ignite/metrics/gan/inception_score.py

```diff
@@ -0,0 +1,123 @@
+
+
+    .. warning::
+
+        Current implementation stores all input data before computing the metric.
+        This can potentially lead to a memory error if the input data is larger than available RAM.
+
+        In distributed configuration, all stored data is mutually collected across all processes
+        using all gather collective operation. This can potentially lead to a memory error.
+
+    Args:
+        splits: number of splits to calculate the mean inception score.
+
+        inception_model: model used for extracting class probabilities from images If not specified,
+            InceptionV3 Pretrained on ImageNet will be used.
+        output_transform: a callable that is used to transform the
+            :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+        device: specifies which device updates are accumulated on. Setting the metric's
+            device to be the same as your ``update`` arguments ensures the ``update`` method is non-blocking. By
+            default, CPU.
+
+    Example:
+
+    .. code-block:: python
+
+        from ignite.metrics.gan import InceptionScore
+
+        m = InceptionScore(splits=4)
+
+        y_pred = torch.rand((16,3,299,299))  # size must match the input shape of the InceptionV3 model
+
+        m.update(y_pred)
+
+        print(m.compute())
+
+    """
+
+    def __init__(
+        self,
+        splits: int = 10,
+        inception_model: Optional[nn.Module] = None,
+        output_transform: Callable = lambda x: x,
+        device: Union[str, torch.device] = torch.device("cpu"),
+    ):
+        if inception_model is None:
+            try:
+                from torchvision import models  # type: ignore
+
+                inception_model = models.inception_v3(pretrained=True, transform_input=False)
+            except ImportError:
+                raise RuntimeError(
+                    "This metric requires torchvision to be installed. "
+                    "You may install torchvision using: \n pip install torchvision \n"
+                )
+        if not isinstance(inception_model, nn.Module):
+            raise TypeError(
+                f"Argument inception_model should be instance of nn.Module, but given {type(inception_model)}"
+            )
+        super(InceptionScore, self).__init__(output_transform=output_transform, device=device)
+        self.inception_model = inception_model.eval().to(self._device)
+        self.n_splits = splits
+
+    @reinit__is_reduced
+    def reset(self) -> None:
+        self._probs = []  # type: List[torch.Tensor]
+
+    @reinit__is_reduced
+    def update(self, output: torch.Tensor) -> None:
+        generated = output.detach()
+        inception_output = self.inception_model(generated)
```

Comment:
We should add `with torch.no_grad()` context manager here to avoid grads creation.